### ISSUE TITLE
FTBFS due to quic-go not supporting go 1.20

### ISSUE THREAD

**Issue by grawlinson** [cloudflare/cloudflared#888]
**Describe the bug**
cloudflared does not compile on go 1.20 due to the quic-go dependency specified in go.mod not supporting 1.20.

As of ~13 hours ago, 1.20 is [supported](https://github.com/quic-go/quic-go/commit/5b72f4c900f209b5705bb0959399d59e495a2c6e) upstream.

**To Reproduce**
Compile with go 1.20 and it will fail with the following error:

```sh
# github.com/lucas-clemente/quic-go/internal/qtls
../pkg/mod/github.com/chungthuang/quic-go@v0.27.1-0.20220809135021-ca330f1dec9f/internal/qtls/go120.go:6:13: cannot use "The version of quic-go you're using can't be built on Go 1.20 yet. For more details, please see https://github.com/lucas-clemente/quic-go/wiki/quic-go-and-Go-versions." (untyped string constant "The version of quic-go you're using can't be built on Go 1.20 yet. F...) as int value in variable declarat
```

The dependency on quic-go should probably be updated sooner rather than later.

**Comment by sudarshan-reddy** [comment]
Thanks for flagging this @grawlinson

**Comment by mraerino** [comment]
this is going to be extra tricky, since cloudflared does not use upstream quic-go, but a fork.
would you accept a PR that changed to using the latest upstream quic-go release?

**Comment by bwesterb** [comment]
We actually do use upstream quic-go, but replace the underlying qtls library. [I prepared one for 1.20](https://github.com/cloudflare/qtls-pq/tree/go1.20) yesterday, so it's ready to go.

**Comment by b0gdanw** [comment]
cloudflared-2023.3.1
go version go1.20.2 darwin/arm64

# github.com/lucas-clemente/quic-go/internal/qtls
vendor/github.com/lucas-clemente/quic-go/internal/qtls/go120.go:6:13: cannot use "The version of quic-go you're using can't be built on Go 1.20 yet. For more details, please see https://github.com/lucas-clemente/quic-go/wiki/quic-go-and-Go-versions." (untyped string constant "The version of quic-go you're using can't be built on Go 1.20 yet. F...) as int value in variable declaration

**Comment by stefanb** [comment]
FYI, go 1.21 will be released soon (see [rc2 announcement](https://groups.google.com/g/golang-announce/c/REcCdQj1taA)), which will make go 1.19 not maintained anymore.

**Comment by AndrewBucklin** [comment]
What’s the latest on this?

**Comment by b0gdanw** [comment]
Successfully built cloudflared 2023.7.1 from source with go version go1.20.6 darwin/arm64

**Comment by i0ntempest** [comment]
Same as above on x86_64

**Comment by stefanb** [comment]
Seems that it was already fixed in [2023.5.1](https://github.com/cloudflare/cloudflared/releases/tag/2023.5.1) via [TUN-7227: Migrate to devincarr/quic-go](https://github.com/cloudflare/cloudflared/commit/9426b603082905d0af8a07bdac866bc1d9c37cba).
